demisto_sdk/commands/common/clients/xsoar_saas/xsoar_saas_api_client.py

```diff
@@ -34,6 +39,35 @@ def is_xsoar_saas(
             and Version(server_version) >= Version(MINIMUM_XSOAR_SAAS_VERSION)
         )
 
+    @classmethod
+    @retry(exceptions=RequestException)
+    def is_xdr_healthy(
+        cls, session: Session, server_config: XsoarSaasClientConfig
+    ) -> bool:
+        """
+        Validates that XDR is healthy.
+
+        Returns:
+            bool: True if SaaS server is healthy, False if not.
+        """
+        url = urljoin(server_config.base_api_url, "public_api/v1/healthcheck")
+        response = session.get(url)
+        response.raise_for_status()
+        try:
+            server_health_status = (response.json().get("status") or "").lower()
+            logger.debug(
+                f"The status of {server_config} health is {server_health_status}"
+            )
+            return (
+                response.status_code == requests.codes.ok
+                and server_health_status == "available"
+            )
+        except JSONDecodeError as e:
+            logger.debug(
+                f"Could not validate if {server_config} is healthy, error:\n{e}"
+            )
+            return False
+
     @validator("session", always=True)
```

Comment:
the use of validators in this class to init the internals is a bit abuse of pydantic. This object isnt meant to be serializable 
We should refactor this to be a normal class that the init method initializes it's parts.

Reply:
done, the clients will not be based on pydantic anymore